src/main/java/com/feedzai/commons/sql/abstraction/engine/AbstractTranslator.java

```diff
@@ -384,6 +387,58 @@ public String translate(final Union union) {
                 .collect(Collectors.joining(delimiter));
     }
 
+    /**
+     * Translates Values.
+     *
+     * @param values a values.
+     * @return values translation.
+     */
+    public String translate(final Values values) {
+        final ArrayList<Values.Row> rows = new ArrayList<>(values.getRows());
+        final String[] aliases = values.getAliases();
+
+        // Given a list of rows and if aliases exist, apply them.
+        if (aliases != null) {
+            rows.forEach(row -> {
+                final List<Expression> expressions = row.getExpressions();
+                for (int i = 0; i < expressions.size() && i < aliases.length; i++) {
+                    expressions.get(i).alias(aliases[i]);
```

Comment:
You are mutating something inside the `values` you are receiving.  
Can this have side-effects?

Reply:
The only side-effect that I can foresee is that if someone defines alias by hand in each row, they will be replaced by the values aliases. But I don't see it as a problem.